src/data/matrix/kronecker.lean

```diff
@@ -256,6 +333,49 @@ kronecker_map_bilinear_mul_mul (algebra.lmul ℕ α).to_linear_map mul_mul_mul_c
   A ⊗ₖ (B ⊗ₖ C) :=
 kronecker_map_assoc₁ _ _ _ _ A B C mul_assoc
 
+lemma trace_kronecker [fintype m] [fintype n] [semiring α]
+  (A : matrix m m α) (B : matrix n n α) :
+  trace (A ⊗ₖ B) = trace A * trace B :=
+trace_kronecker_map_bilinear (algebra.lmul ℕ α).to_linear_map _ _
+
+lemma det_kronecker [fintype m] [fintype n] [decidable_eq m] [decidable_eq n] [comm_ring R]
+  (A : matrix m m R) (B : matrix n n R) :
+  det (A ⊗ₖ B) = det A ^ fintype.card n * det B ^ fintype.card m :=
+begin
+  refine
+    (det_kronecker_map_bilinear (algebra.lmul ℕ R).to_linear_map mul_mul_mul_comm _ _).trans _,
+  congr' 3,
+  { ext i j, exact mul_one _},
+  { ext i j, exact one_mul _},
+end
+
+lemma inv_kronecker [fintype m] [fintype n] [decidable_eq m] [decidable_eq n] [comm_ring R]
+  (A : matrix m m R) (B : matrix n n R) :
+  (A ⊗ₖ B)⁻¹ = (A⁻¹ ⊗ₖ B⁻¹) :=
+begin
+  -- handle the special cases where either matrix is not invertible
+  by_cases hA : is_unit A.det, swap,
```

Comment:
Should the `swap` be on another line?

Reply:
I don't know what the rules are for swap. Usually the rule is "start a `{}` block as soon as there is a new goal", but obviously that doesn't work for swap. IMO by putting this on the same line it makes it clearer that we're not done arranging the subgoals